src/pluginmain.ts

```diff
@@ -682,4 +709,35 @@ export class MyPlugin {
 
         this.updateCounterText();
     }
+
+    // find the nearest other point. takes an object with targetPoint or
+    // polygon and point index. Points in the given polygon is excluded from search.
+    private getNearestPoint({
+        polygon: targetPolygon,
+        index,
+        point: targetPoint = targetPolygon.getPath().getAt(index)
+    }: any) {
```

Comment:
Why are we packing all the parameters into a single { }? 
Could we pass the parameters like: 
```
getNearestPoint(polygon, index) { 
    targetPoint = polygon.getPath().getAt(index)
```

Reply:
I find it easier to work with a single param while developing as I can more easily use or modify API without worrying about getting the param ordering wrong. Since this method wouldn't be called extremely often, I don't think there's a considerable performance penalty. I made `point` an optional input with a default value because I was planning for the case where the polygon hasn't been created yet.